api/project.py

```
@@ -0,0 +1,112 @@
+
+    def analyze_sources(self, ignore: Optional[List[str]] = None) -> Graph:
+        if ignore is None:
+            ignore = []
+        analyzer = SourceAnalyzer()
+        analyzer.analyze_local_folder(self.path, self.graph, ignore)
+
+        try:
+            # Save processed commit hash to the DB
+            repo = Repo(self.path)
+            current_commit = repo.head.commit
+            set_repo_commit(self.name, current_commit.hexsha)
```

Ruff (0.8.2)

`set_repo_commit` may be undefined, or defined from star imports

(F405)